iree/compiler/Codegen/Common/ConvertToDestinationPassingStylePass.cpp

```diff
@@ -379,6 +379,40 @@ struct AdaptLinalgInputOperandToOutputOperand
     return success();
   }
 };
+
+struct RemoveCstOutsDependency
+    : public OpInterfaceRewritePattern<linalg::LinalgOp> {
+  using OpInterfaceRewritePattern<linalg::LinalgOp>::OpInterfaceRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg::LinalgOp op,
+                                PatternRewriter &rewriter) const override {
+    rewriter.startRootUpdate(op);
+    bool modifiedOutput = false;
+    Location loc = op.getLoc();
+    for (OpOperand *opOperand : op.getOutputOperands()) {
```

Comment:
I dont think this is necessarily related to splat constant. The condition I see required here are that 
1) The operand "comes from a constant", i.e. you walk through ops (like say casts and reshapes) you reach a constant
2) The outs operand is not used in the body.

Then you can just replace this with an init tensor. I dont think you need the fill...

Reply:
The outs operand is used in the body because this is a reduction op.

Is there a copy op in tensors? I check if it is a splat constant because I only find linalg.fill op. Maybe I can create a generic op if we want to handle non-splat constant cases.

Reply:
Why is this pattern needed at all? When the original LinalgOp is processed during bufferization, it will figure out that the constant is not writable. At that point, bufferization will allocate a new buffer. This is identical to `linalg.init_tensor`.

Reply:
But we don't want the buffer to be created. With this pattern it can be made destination style passing.

Reply:
I thought the `init_tensor` elimination was deprecated. We dont use it in IREE, but if you think that can help here would be worth a try....

Reply:
Just saying, if you don't remove the `init_tensor` before it hits bufferization, you will have a `memref.alloc` in the bufferized code.

Reply:
I understand... I know that this happens, but it means we have to have some fairly "ad-hoc" transformations to get things into destination passing style. We can get by, but maybe `init_tensor` always allocates is not a sustainable solution....

Reply:
With this pattern, we'll create `linalg.init_tensor` + `linalg.fill`. The `linalg.init_tensor` will be replaced with IREE `flow.dispatch.tensor.load` (in the same pass), which avoids alloca ops.